#! /bin/sh /usr/share/dpatch/dpatch-run
## 18_irman_fix.dpatch by Tristan Hill <stan@saticed.me.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Irman decode fix from CVS HEAD

@DPATCH@

--- lirc-0.8.2~/daemons/ir_remote.c	2007-05-06 10:46:59.000000000 +0100
+++ lirc-0.8.2/daemons/ir_remote.c	2007-08-12 03:33:21.429361048 +0100
@@ -190,11 +190,11 @@
 		LOGPRINTF(1,"post");
 	}
 
-	all = pre;
+	all = (pre&gen_mask(remote->pre_data_bits));
 	all <<= remote->bits;
-	all |= code;
+	all |= is_raw(remote) ? code:(code&gen_mask(remote->bits));
 	all <<= remote->post_data_bits;
-	all |= post;
+	all |= post&gen_mask(remote->post_data_bits);
 	
 	toggle_bit_mask_state = all&remote->toggle_bit_mask;
 
